File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        left, right, res = 0, len(numbers) - 1, [-1 , -1]

        while left < right:
            if numbers[left] + numbers[right] > target:
                right -= 1
            elif numbers[left] + numbers[right] < target:
                left += 1
            else:
                res = [left + 1, right + 1]
                break
        
        return res

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {
    int left = 0, right = numbersSize - 1;
    int* res = (int*)malloc(2 * sizeof(int));
    *returnSize = 2;

    res[0] = res[1] = -1;

    while(left < right) {
        if(numbers[left] + numbers[right] > target) right--;
        else if(numbers[left] + numbers[right] < target) left++;
        else {
            res[0] = left + 1;
            res[1] = right + 1;
            break;
        }
    }

    return res;
}